python/lsst/ts/standardscripts/maintel/closed_loop.py

```diff
@@ -0,0 +1,519 @@
+
+    Parameters
+    ----------
+    index : `int`, optional
+        Index of Script SAL component (default=1).
+    remotes : `bool`, optional
+        Should the remotes be created (default=True)? For unit testing this
+        can be set to False, which allows one to mock the remotes behaviour.
+    descr : `str`, optional
+        Short description of the script.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    - "Taking image...": If taking in-focus detection image.
+    - "[N/MAX_ITER]: CWFS loop starting...": Before each cwfs iteration, where
+        "N"  is the iteration number and "MAX_ITER" is the maximum number of
+        iterations.
+    - "[N/MAX_ITER]: CWFS converged.": Once CWFS reaches convergence.
+    - "[N/MAX_ITER]: CWFS applying correction.": Just before
+        corrections are applied.
+
+    **Details**
+
+    This script is used to perform measurements of the wavefront error, then
+    propose dof offsets based on ts_ofc. The offsets are not applied
+    automatically and must be turned on by the user through
+    apply_corrections attribute.
+    """
+
+    def __init__(self, index=1, add_remotes=True, descr=""):
+        super().__init__(
+            index=index,
+            descr=descr,
+        )
+
+        self.mtaos = RemoteGroup(
+            domain=self.domain,
+            components=["MTAOS"],
+            intended_usage=None if add_remotes else Usages.DryTest,
+            log=self.log,
+        )
+
+        mtcs_usage = None if add_remotes else MTCSUsages.DryTest
+        self.mtcs = MTCS(self.domain, intended_usage=mtcs_usage, log=self.log)
+
+        lsstcam_usage = None if add_remotes else ComCamUsages.StateTransition
+        self.lsstcam = ComCam(
+            self.domain,
+            intended_usage=lsstcam_usage,
+            log=self.log,
+            tcs_ready_to_take_data=self.mtcs.ready_to_take_data,
+        )
+
+        # The following attributes are set via the configuration:
+        self.filter = None
+        self.grating = None
+
+        # exposure time for the intra/extra images (in seconds)
+        self.exposure_time = None
+
+        # end of configurable attributes
+
+    @classmethod
+    def get_schema(cls) -> typing.Dict[str, typing.Any]:
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            $id: https://github.com/lsst-ts/ts_standardscripts/auxtel/ClosedLoopCwfs.yaml
+            title: ClosedLoopCwfs v1
+            description: Configuration for ClosedLoopCwfs Script.
+            type: object
+            properties:
+              mode:
+                description: >-
+                    Mode to use for the script. If set to "cwfs", the script will
+                    use the corner wavefront sensors, if set to "fam" the script
+                    will use the full array mode.
+                type: string
+                enum: ["cwfs", "fam"]
+                default: cwfs
+              rotation_angle:
```

Comment:
Why do you need the user to provide this? you should get it from the system. However I am not even sure you need to get it as this information is available in the image header and the MTAOS should be able to determine this.